src/stratis_cli/_actions/_top.py

```diff
@@ -538,3 +602,139 @@ def add_cache_devices(namespace):  # pylint: disable=too-many-locals
                 )
                 % (namespace.pool_name, ", ".join(devnodes_added), ", ".join(blockdevs))
             )
+
+    @staticmethod
+    def add_key(namespace):
+        """
+        Add key to kernel keyring.
+
+        :raises StratisCliEngineError:
+        :raises StratisCliPropertyNotFoundError:
+        :raises StratisCliNameConflictError:
+        :raises StratisCliIncoherenceError:
+        """
+        proxy = get_object(TOP_OBJECT)
+
+        key_list = fetch_keylist_property(proxy)
+        if namespace.keydesc in key_list:
+            raise StratisCliNameConflictError("key", namespace.keydesc)
+
+        ((changed, existing_modified), return_code, message) = add_update_key(
+            proxy,
+            namespace.keydesc,
+            namespace.capture_key,
+            keyfile_path=namespace.keyfile_path,
+        )
+
+        if return_code != StratisdErrors.OK:  # pragma: no cover
+            raise StratisCliEngineError(return_code, message)
+
+        if not changed and not existing_modified:  # pragma: no cover
+            raise StratisCliIncoherenceError(
+                (
+                    "Key %s was reported to not exist but stratisd reported "
+                    "that no change was made to the key"
+                )
+                % namespace.keydesc
+            )
+
+        # A key was updated even though there was no key reported to already exist.
+        if changed and existing_modified:  # pragma: no cover
+            raise StratisCliIncoherenceError(
+                (
+                    "Key %s was reported to not exist but stratisd reported "
+                    "that it reset an existing key"
+                )
+                % namespace.keydesc
+            )
+
+    @staticmethod
+    def update_key(namespace):
+        """
+        Update an existing key in the kernel keyring.
+
+        :raises StratisCliEngineError:
+        :raises StratisCliResourceNotFoundError:
+        :raises StratisCliPropertyNotFoundError:
+        :raises StratisCliIncoherenceError:
+        """
+        proxy = get_object(TOP_OBJECT)
+
+        key_list = fetch_keylist_property(proxy)
+        if namespace.keydesc not in key_list:
+            raise StratisCliResourceNotFoundError("reset", namespace.keydesc)
+
+        ((changed, existing_modified), return_code, message) = add_update_key(
+            proxy,
+            namespace.keydesc,
+            namespace.capture_key,
+            keyfile_path=namespace.keyfile_path,
+        )
+
+        if return_code != StratisdErrors.OK:  # pragma: no cover
```

Comment:
You can remove the "no cover" pragma here because it is generally testable via the too long key.